### PR TITLE
docs(certificates): Updates the procedures for user-supplied CA certs

### PR DESCRIPTION
**Pull request by PaulRMellor**
Signed-off-by: prmellor <pmellor@redhat.com>

**Documentation**

- Creates a new assembly for user-specified certificates
- Updates the procedures to make clear what is being replaced or renewed
- Removes the setp to delete the existing secrets when installing custom  certificates 
- Makes a few name consistency updates for the clients CA 

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: documentation/assemblies/security/assembly-using-your-own-certificates.adoc [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+= Using your own CA certificates and private keys
+
+[role="_abstract"]
+Install and use your own CA certificates and private keys instead of using the defaults generated by the Cluster Operator.
+You can replace the cluster and clients CA certificates and private keys.
+
+You can switch to using your own CA certificates and private keys in the following ways:
+
+* Install your own CA certificates and private keys before deploying your Kafka cluster
+* Replace the default CA certificates and private keys with your own after deploying a Kafka cluster
```
Comment (scholzj) [inline documentation/assemblies/security/assembly-using-your-own-certificates.adoc:15]: Should we make it clear that the replacing happens essentially through the `Renew CA certificates and private keys`process?

Comment (PaulRMellor) [reply documentation/assemblies/security/assembly-using-your-own-certificates.adoc:15]: We had something on line 25. I've moved that up and added something to the bullet too

**Review by scholzj — APPROVED** [review]

File: documentation/modules/security/proc-installing-your-own-ca-certificates.adoc [hunk 2]
```diff
@@ -36,73 +38,38 @@ The chain should be in the following order:
...
-kubectl delete secret _CA-KEY-SECRET_
-+
-_CA-KEY-SECRET_ is the name of CA key:
 +
-* `_CLUSTER-NAME_-cluster-ca` for the cluster CA key
-* `_CLUSTER-NAME_-clients-ca` for the clients CA key
+Replace _<cluster_name>_ with the name of your Kafka cluster.
 
-.. Create the new secret.
+. Create a new secret to replace the private key generated by the Cluster Operator.
```
Comment (ppatierno) [inline documentation/modules/security/proc-installing-your-own-ca-certificates.adoc:72]: Isn't this procedure about installing your own CA cert + key from scratch with cluster not running? so we are not replacing, are't we?

File: documentation/modules/security/proc-installing-your-own-ca-certificates.adoc [hunk 3]
```diff
@@ -137,7 +104,6 @@ kubectl annotate secret _CA-KEY-SECRET_ strimzi.io/ca-key-generation=_CA-KEY-GEN
 * Annotation `strimzi.io/ca-cert-generation=_CA-CERTIFICATE-GENERATION_` defines the generation of a new CA certificate.
 * Annotation `strimzi.io/ca-key-generation=_CA-KEY-GENERATION_` defines the generation of a new CA key.
 +
-If you are replacing CA certificates automatically generated by the Cluster Operator, use the next higher incremental value from the existing annotation and follow the xref:proc-replacing-your-own-private-keys-{context}[replacing CA keys procedure].
 If there are no CA certificates automatically generated by the Cluster Operator, start from 0 (zero) as the incremental value (`strimzi.io/ca-cert-generation=0`) for your own CA certificate. Set a higher incremental value when you renew the certificates.
```
Comment (ppatierno) [inline documentation/modules/security/proc-installing-your-own-ca-certificates.adoc:107]: `If there are no CA certificates automatically generated by the Cluster Operator` ... if this is about installing your own cert + key from scratch, this is always true, or? So we could omit it and say to start from 0.

File: documentation/modules/security/proc-installing-your-own-ca-certificates.adoc (hunk 2 above)
Comment (PaulRMellor) [reply documentation/modules/security/proc-installing-your-own-ca-certificates.adoc:72]: Yes. That could be misread. I've changed as follows:

> 1. Create a new secret that contains the CA certificate.
>  2. Create a new secret that contains the private key.

File: documentation/modules/security/proc-installing-your-own-ca-certificates.adoc (hunk 3 above)
Comment (PaulRMellor) [reply documentation/modules/security/proc-installing-your-own-ca-certificates.adoc:107]: Well, I guess you could just be using your own cluster certificates and keeping the clients generated ones?

Comment (ppatierno) [reply documentation/modules/security/proc-installing-your-own-ca-certificates.adoc:107]: Yeah that's possible but they are following two separated generations. They are not sharing the generation value. So this sentence is not so clear to me imho.

Comment (PaulRMellor) [reply documentation/modules/security/proc-installing-your-own-ca-certificates.adoc:107]: Yeah. I think it's clearer without the first sentence.

> Start from 0 (zero) as the incremental value (`strimzi.io/ca-cert-generation=0`) for your own CA certificate. Set a higher incremental value when you renew the certificates.

**Pull request merged**